### PR TITLE
Prevent release with critical failures

### PR DESCRIPTION
**Pull request by WhitShake**
User will be notified via popup if Fatal diagnostics or open conversations are present when attempting approval. Issue checkboxes must be checked to override and continue approval.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Client/css/shared/codeline.scss [hunk 1]
```diff
@@ -355,23 +355,58 @@ code {
...
-.code-diagnostics-error {
-    background-color: var(--code-diagnostics-error-color);
-    border: 1px solid red;
+$diagnostics: ( 
+    fatal: var(--code-diagnostics-fatal-color), 
+    error: var(--code-diagnostics-error-color), 
+    warn: var(--code-diagnostics-warn-color), 
+    info: var(--code-diagnostics-info-color) );
+
+@each $type, $color in $diagnostics {
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Client/css/shared/codeline.scss:364]: Interesting I didn't know this was possible.

File: src/dotnet/APIView/APIViewWeb/Client/css/shared/codeline.scss [hunk 2]
```diff
@@ -355,23 +355,58 @@ code {
...
+    color: #0d1dfc;
+
+    &:visited {
+        color: #3d4afd;
+    }
+
+    &:active,
+    &:hover {
+        color: #00CBEB;
+    }
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Client/css/shared/codeline.scss:409]: Why not declare the colors as variables in `theme-colors.scss`? Better to have themes specification all in one place.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 3]
```diff
@@ -37,6 +38,20 @@
...
+                        @count
+                        <span class="visually-hidden">unread messages</span>
+                    </span>
+                }
+                else if (Model.ReviewContent.ActiveConversationsInActiveAPIRevision > 0)
+                {
+                    var count = (Model.ReviewContent.ActiveConversationsInActiveAPIRevision > 10) ? "9+" : $"{Model.ReviewContent.ActiveConversationsInActiveAPIRevision}";
+                    <span class="position-absolute start translate-middle badge rounded-pill bg-danger" id="jump-to-first-comment">
+                    </span>
+                }
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:54]: This doesn't look like new code. Not sure what's happening here

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs [hunk 4]
```diff
@@ -72,6 +75,9 @@ public class ReviewPageModel : PageModel
         public string NotificationMessage { get; set; }
         public UserPreferenceModel UserPreference { get; set; }
 
+
+        public CodeFile APIView { get; set; }
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs:79]: What is this for?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs [hunk 5]
```diff
@@ -17,6 +17,9 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.Extensions.Configuration;
 using Microsoft.TeamFoundation.Common;
+using MongoDB.Driver;
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs:20]: Why MongoDb Driver?

**Review by chidozieononiwu — CHANGES_REQUESTED** [review]
Lets deploy to staging to test it out. Also seems like the razor pages got screwed up a bit. Happens at time will need to take a closer look.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs (hunk 4 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs:79]: Should have been deleted, ty. Fixed

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml (hunk 3 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:54]: Definitely not new code, not sure why it's coming through as new.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs (hunk 5 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs:20]: This is also not something I added, at least not intentionally. Tested by commenting it out, still works. Will delete for now.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml (hunk 3 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:54]: Remove this code.

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**